From ff3c623cf1fbd681bc123417da4a4c23390ceb7b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pablo=20Correa=20G=C3=B3mez?= <ablocorrea@hotmail.com>
Date: Sun, 18 Jul 2021 20:53:01 +0200
Subject: [PATCH] Hide and disable building printer panel

The printer panel requires `libcups`, which is known to crash
in Alpine: https://gitlab.alpinelinux.org/alpine/aports/-/issues/11937
Opening the printer panel crashes g-c-c and prevents it from
opening again. Hide and do not build the panel until the upstream
issue is fixed
---
 debian/patches/pureos/shell-Hide-some-panels-on-phones.patch | 3 +--
 panels/meson.build                                           | 1 -
 shell/cc-panel-loader.c                                      | 2 --
 tests/meson.build                                            | 1 -
 4 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/debian/patches/pureos/shell-Hide-some-panels-on-phones.patch b/debian/patches/pureos/shell-Hide-some-panels-on-phones.patch
index 3fd0eac5c..3a17401fb 100644
--- a/debian/patches/pureos/shell-Hide-some-panels-on-phones.patch
+++ b/debian/patches/pureos/shell-Hide-some-panels-on-phones.patch
@@ -33,7 +33,7 @@ new file mode 100644
 index 0000000..e8a446c
 --- /dev/null
 +++ b/shell/phone-panels.h
-@@ -0,0 +1,52 @@
+@@ -0,0 +1,51 @@
 +static CcPanelLoaderVtable phone_panels[] =
 +  {
 +   /* PANEL_TYPE("applications",     cc_applications_panel_get_type,         NULL), */
@@ -60,7 +60,6 @@ index 0000000..e8a446c
 +   PANEL_TYPE("notifications",    cc_notifications_panel_get_type,        NULL),
 +   PANEL_TYPE("online-accounts",  cc_goa_panel_get_type,                  NULL),
 +   PANEL_TYPE("power",            cc_power_panel_get_type,                NULL),
-+   PANEL_TYPE("printers",         cc_printers_panel_get_type,             NULL),
 +   PANEL_TYPE("region",           cc_region_panel_get_type,               NULL),
 +   /* PANEL_TYPE("removable-media",  cc_removable_media_panel_get_type,      NULL), */
 +   PANEL_TYPE("search",           cc_search_panel_get_type,               NULL),
diff --git a/panels/meson.build b/panels/meson.build
index 2f4fdc5e3..7113a9867 100644
--- a/panels/meson.build
+++ b/panels/meson.build
@@ -18,7 +18,6 @@ panels = [
   'notifications',
   'online-accounts',
   'power',
-  'printers',
   'region',
   'removable-media',
   'search',
diff --git a/shell/cc-panel-loader.c b/shell/cc-panel-loader.c
index f20384394..9c329a477 100644
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -50,7 +50,6 @@ extern GType cc_wifi_panel_get_type (void);
 extern GType cc_notifications_panel_get_type (void);
 extern GType cc_goa_panel_get_type (void);
 extern GType cc_power_panel_get_type (void);
-extern GType cc_printers_panel_get_type (void);
 extern GType cc_region_panel_get_type (void);
 extern GType cc_removable_media_panel_get_type (void);
 extern GType cc_search_panel_get_type (void);
@@ -114,7 +113,6 @@ static CcPanelLoaderVtable default_panels[] =
   PANEL_TYPE("notifications",    cc_notifications_panel_get_type,        NULL),
   PANEL_TYPE("online-accounts",  cc_goa_panel_get_type,                  NULL),
   PANEL_TYPE("power",            cc_power_panel_get_type,                NULL),
-  PANEL_TYPE("printers",         cc_printers_panel_get_type,             NULL),
   PANEL_TYPE("region",           cc_region_panel_get_type,               NULL),
   PANEL_TYPE("removable-media",  cc_removable_media_panel_get_type,      NULL),
   PANEL_TYPE("search",           cc_search_panel_get_type,               NULL),
diff --git a/tests/meson.build b/tests/meson.build
index d4fe361ef..01f7c923c 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -6,5 +6,4 @@ endif
 
 subdir('interactive-panels')
 
-subdir('printers')
 subdir('info')
-- 
2.17.1

